### ISSUE TITLE
paddlex的evaluate接口问题

### ISSUE THREAD

**Issue by axdo** [PaddlePaddle/PaddleX#655]
问题类型：其它  
**问题描述**  

========================

请在这里描述您的问题
如图
![ZO CKT3GL)IUBWHIKM%8`II](https://user-images.githubusercontent.com/46195295/113234265-84480800-92d3-11eb-858a-11c9affc884c.png)
请问是咋回事？

**Comment by FlyingQianMM** [comment]
麻烦提供下评估脚本代码、对应的paddlex和paddle版本

**Comment by axdo** [comment]
> 麻烦提供下评估脚本代码、对应的paddlex和paddle版本

import os
os.environ['CUDA_VISIBLE_DEVICES'] = '0'

from paddlex.det import transforms
import paddlex as pdx


# 定义训练和验证时的transforms
# API说明 https://paddlex.readthedocs.io/zh_CN/develop/apis/transforms/det_transforms.html
train_transforms = transforms.Compose([
    transforms.MixupImage(mixup_epoch=250),
    transforms.RandomDistort(),
    transforms.RandomExpand(),
    transforms.RandomCrop(),
    transforms.ResizeByShort(short_size=800, max_size=1333),
    transforms.Resize(
        target_size=608, interp='RANDOM'),
    transforms.RandomHorizontalFlip(),
    transforms.Normalize(),
])

eval_transforms = transforms.Compose([
    transforms.ResizeByShort(short_size=800, max_size=1333),
    transforms.Resize(
        target_size=608, interp='CUBIC'),
    transforms.Normalize(),
])

# 定义训练和验证所用的数据集
# API说明：https://paddlex.readthedocs.io/zh_CN/develop/apis/datasets.html#paddlex-datasets-vocdetection
train_dataset = pdx.datasets.VOCDetection(
    data_dir='./data/data77544/D0002',
    file_list='./data/data77544/D0002/train_list.txt',
    label_list='./data/data77544/D0002/labels.txt',
    transforms=train_transforms,
    shuffle=True)
eval_dataset = pdx.datasets.VOCDetection(
    data_dir='./data/data77544/D0002',
    file_list='./data/data77544/D0002/val_list.txt',
    label_list='./data/data77544/D0002/labels.txt',
    transforms=eval_transforms)

# 初始化模型，并进行训练
# 可使用VisualDL查看训练指标，参考https://paddlex.readthedocs.io/zh_CN/develop/train/visualdl.html
num_classes = len(train_dataset.labels)

# API说明: https://paddlex.readthedocs.io/zh_CN/develop/apis/models/detection.html#paddlex-det-yolov3
model = pdx.det.YOLOv3(num_classes=num_classes, backbone='MobileNetV1')
model.evaluate(eval_dataset)

**Comment by axdo** [comment]
> 麻烦提供下评估脚本代码、对应的paddlex和paddle版本

import os
os.environ['CUDA_VISIBLE_DEVICES'] = '0'

from paddlex.det import transforms
import paddlex as pdx
train_transforms = transforms.Compose([
    transforms.MixupImage(mixup_epoch=250),
    transforms.RandomDistort(),
    transforms.RandomExpand(),
    transforms.RandomCrop(),
    transforms.ResizeByShort(short_size=800, max_size=1333),
    transforms.Resize(
        target_size=608, interp='RANDOM'),
    transforms.RandomHorizontalFlip(),
    transforms.Normalize(),
])

eval_transforms = transforms.Compose([
    transforms.ResizeByShort(short_size=800, max_size=1333),
    transforms.Resize(
        target_size=608, interp='CUBIC'),
    transforms.Normalize(),
])
train_dataset = pdx.datasets.VOCDetection(
    data_dir='./data/data77544/D0002',
    file_list='./data/data77544/D0002/train_list.txt',
    label_list='./data/data77544/D0002/labels.txt',
    transforms=train_transforms,
    shuffle=True)
eval_dataset = pdx.datasets.VOCDetection(
    data_dir='./data/data77544/D0002',
    file_list='./data/data77544/D0002/val_list.txt',
    label_list='./data/data77544/D0002/labels.txt',
    transforms=eval_transforms)
num_classes = len(train_dataset.labels)
model = pdx.det.YOLOv3(num_classes=num_classes, backbone='MobileNetV1')
model.evaluate(eval_dataset)

ai studio  paddle2.0 paddlex1.3.7最新都是

**Comment by FlyingQianMM** [comment]
> model = pdx.det.YOLOv3(num_classes=num_classes, backbone='MobileNetV1')
> model.evaluate(eval_dataset)

模型还未训练是无法进行evalute的，如果是要加载已经训练好的模型进行evaluate：
```
model = pdx.load_model(xxxx)
model.evaluate(eval_dataset)
```
xxxx为模型文件夹所在路径

**Comment by axdo** [comment]
> > model = pdx.det.YOLOv3(num_classes=num_classes, backbone='MobileNetV1')
> > model.evaluate(eval_dataset)
> 
> 模型还未训练是无法进行evalute的，如果是要加载已经训练好的模型进行evaluate：
> 
> ```
> model = pdx.load_model(xxxx)
> model.evaluate(eval_dataset)
> ```
> 
> xxxx为模型文件夹所在路径

嗷嗷 您好顺便想请问一下加载自己的预训练模型在训练就是
`model = pdx.load_model(xxxx)
model.train()` xxx是训练好的模型文件夹
是这样吗？

**Comment by axdo** [comment]
> > model = pdx.det.YOLOv3(num_classes=num_classes, backbone='MobileNetV1')
> > model.evaluate(eval_dataset)
> 
> 模型还未训练是无法进行evalute的，如果是要加载已经训练好的模型进行evaluate：
> 
> ```
> model = pdx.load_model(xxxx)
> model.evaluate(eval_dataset)
> ```
> 
> xxxx为模型文件夹所在路径

还有这个evaluate完了就啥也没有了 结果保存在哪里了？

**Comment by FlyingQianMM** [comment]
> 还有这个evaluate完了就啥也没有了 结果保存在哪里了？

打印下evaluate()的返回值

> 嗷嗷 您好顺便想请问一下加载自己的预训练模型在训练就是

指定train()的pretrain_weight就可以了，具体看下文档说明
